lighthouse-core/audits/seo/font-size.js

```diff
@@ -121,64 +121,70 @@ function nodeToTableNode(node) {
-    const stylesheet = styleDeclaration.stylesheet;
-
-    if (stylesheet) {
-      let source;
-      const selector = rule.selectors.map(item => item.text).join(', ');
-
-      if (stylesheet.sourceURL) {
-        const url = new URL(stylesheet.sourceURL, baseURL);
-        const range = styleDeclaration.range;
-        source = `${url.href}`;
+    selector = rule.selectors.map(item => item.text).join(', ');
+  }
 
-        if (range) {
-          // `stylesheet` can be either an external file (stylesheet.startLine will always be 0),
-          // or a <style> block (stylesheet.startLine will vary)
-          const absoluteStartLine = range.startLine + stylesheet.startLine + 1;
-          const absoluteStartColumn = range.startColumn + stylesheet.startColumn + 1;
+  if (styleDeclaration.stylesheet && !styleDeclaration.stylesheet.sourceURL) {
+    return {
+      source: {type: 'code', value: 'dynamic'},
+      selector,
+    };
+  }
 
-          source += `:${absoluteStartLine}:${absoluteStartColumn}`;
-        }
-      } else {
-        // dynamically injected to page
-        source = 'dynamic';
+  if (styleDeclaration.range) {
+    // The magic comment sourceURL, if any. Used to resolve a url relative to the baseUrl.
+    // Note, URLs resolved from a magic comment aren't expected to _actually_ exist ...
+    const sourceURL = styleDeclaration.stylesheet ? styleDeclaration.stylesheet.sourceURL : '';
+    const url = new URL(sourceURL, baseURL).href;
+    // TODO when is `styleDeclaration.range` null ???
+    let {startLine, startColumn} = styleDeclaration.range ? styleDeclaration.range : {startLine: 0, startColumn: 0};
+
+    // Inline elements need to add the startLine/startColumn of the <script> element to the ui location, so that the location
+    // is relevant to the HTML file the stylesheet is within.
+    // But, if an inline stylesheet has a sourceURL magic comment, `hasSourceURL` is true and thus `styleDeclaration.range` is sufficient.
+    if (styleDeclaration.stylesheet && styleDeclaration.stylesheet.isInline && !styleDeclaration.stylesheet.hasSourceURL) {
+      startLine += styleDeclaration.stylesheet.startLine;
+      // The column the stylesheet begins on is only relevant if the rule is declared on the same line.
+      // `<div>...</div><style>.some-rule{}`
+      if (styleDeclaration.range && styleDeclaration.range.startLine === 0) {
+        startColumn += styleDeclaration.stylesheet.startColumn;
       }
-
-      return {
-        selector,
-        source,
-      };
     }
+
+    const magicCommentSourceUrl = styleDeclaration.stylesheet && styleDeclaration.stylesheet.hasSourceURL ? sourceURL : undefined;
+    return {
+      source: {type: 'ui-location', sourceURL: magicCommentSourceUrl, url, line: startLine, column: startColumn},
+      selector,
+    };
   }
 
   return {
-    selector: '',
-    source: 'Unknown',
+    selector,
+    source: {type: 'code', value: 'Unknown'},
```

Comment:
curious what would be unknown. anyone know how to query a database of LHRs? @exterkamp @patrickhulce

Reply:
https://discuss.httparchive.org/t/analyzing-lighthouse-scores-across-the-web/1600